modules/siddhi-core/src/main/java/io/siddhi/core/util/cache/CacheUtils.java

```diff
@@ -0,0 +1,99 @@
+import io.siddhi.core.util.collection.executor.CompareCollectionExecutor;
+import io.siddhi.core.util.collection.operator.CompiledCondition;
+import io.siddhi.core.util.collection.operator.OverwriteTableIndexOperator;
+
+import java.util.List;
+
+/**
+ * class containing utils related to store table cache
+ */
+public class CacheUtils {
+    public static int findEventChunkSize(StreamEvent streamEvent) {
+        int chunkSize = 1;
+        if (streamEvent == null) {
+            return 0;
+        }
+        StreamEvent streamEventCopy = streamEvent;
+        while (streamEventCopy.hasNext()) {
+            chunkSize = chunkSize + 1;
+            streamEventCopy = streamEventCopy.getNext();
+        }
+        return chunkSize;
+    }
+
+    public static String getPrimaryKey(CompiledCondition compiledCondition, StateEvent matchingEvent) {
+        StringBuilder primaryKey = new StringBuilder();
+        StreamEvent streamEvent = matchingEvent.getStreamEvent(0);
+        Object[] data = null;
+        if (streamEvent != null) {
+            data = streamEvent.getOutputData();
+        }
+
+        if (compiledCondition instanceof OverwriteTableIndexOperator) {
+            OverwriteTableIndexOperator operator = (OverwriteTableIndexOperator) compiledCondition;
+            if (operator.getCollectionExecutor() instanceof AndMultiPrimaryKeyCollectionExecutor) {
+                AndMultiPrimaryKeyCollectionExecutor executor = (AndMultiPrimaryKeyCollectionExecutor) operator.
+                        getCollectionExecutor();
+                List<ExpressionExecutor> lis = executor.getMultiPrimaryKeyExpressionExecutors();
+                for (ExpressionExecutor ee : lis) {
+                    if (ee instanceof VariableExpressionExecutor) {
+                        VariableExpressionExecutor vee = (VariableExpressionExecutor) ee;
+                        primaryKey.append(data[vee.getPosition()[3]]);
+                        primaryKey.append(":-:");
+                    }
+                }
+            } else if (operator.getCollectionExecutor() instanceof CompareCollectionExecutor) {
+                CompareCollectionExecutor executor = (CompareCollectionExecutor) operator.getCollectionExecutor();
+                if (executor.getValueExpressionExecutor() instanceof ConstantExpressionExecutor) {
+                    primaryKey.append(((ConstantExpressionExecutor) executor.getValueExpressionExecutor()).getValue());
+                } else if (executor.getValueExpressionExecutor() instanceof VariableExpressionExecutor) {
+                    VariableExpressionExecutor vee = (VariableExpressionExecutor) executor.getValueExpressionExecutor();
+                    primaryKey.append(data[vee.getPosition()[3]]);
+                } else {
+                    return null;
+                }
+            }
+        }
+        return primaryKey.toString();
+    }
+
+    public static String getPrimaryKeyFromMatchingEvent(StateEvent matchingEvent) {
```

Comment:
Can't find where this is used too?

Reply:
will remove for the same reason mentioned above